control/phaseplot.py

```diff
@@ -1,61 +1,935 @@
+    _update = lambda t, x, u, params: sys(t, x, *params.get('args', ()))
+    _output = lambda t, x, u, params: np.array([])
+    return NonlinearIOSystem(
+        _update, _output, states=2, inputs=0, outputs=0, name="_callable")
 
+# Set axis limits for the plot
+def _set_axis_limits(ax, pointdata):
+    # Get the current axis limits
+    if ax.lines:
+        xlim, ylim = ax.get_xlim(), ax.get_ylim()
+    else:
+        # Nothing on the plot => always use new limits
+        xlim, ylim = [np.inf, -np.inf], [np.inf, -np.inf]
 
-def _find(condition):
-    """Returns indices where ravel(a) is true.
-    Private implementation of deprecated matplotlib.mlab.find
-    """
-    return np.nonzero(np.ravel(condition))[0]
+    # Short utility function for updating axis limits
+    def _update_limits(cur, new):
+        return [min(cur[0], np.min(new)), max(cur[1], np.max(new))]
+
+    # If we were passed a box, use that to update the limits
+    if isinstance(pointdata, list) and len(pointdata) == 4:
+        xlim = _update_limits(xlim, [pointdata[0], pointdata[1]])
+        ylim = _update_limits(ylim, [pointdata[2], pointdata[3]])
+
+    elif isinstance(pointdata, np.ndarray):
+        pointdata = np.atleast_2d(pointdata)
+        xlim = _update_limits(
+            xlim, [np.min(pointdata[:, 0]), np.max(pointdata[:, 0])])
+        ylim = _update_limits(
+            ylim, [np.min(pointdata[:, 1]), np.max(pointdata[:, 1])])
+
+    # Keep track of the largest dimension on the plot
+    maxlim = max(xlim[1] - xlim[0], ylim[1] - ylim[0])
+
+    # Set the new limits
+    ax.autoscale(enable=True, axis='x', tight=True)
+    ax.autoscale(enable=True, axis='y', tight=True)
+    ax.set_xlim(xlim)
+    ax.set_ylim(ylim)
+
+    return xlim, ylim, maxlim
+
+
+# Find equilibrium points
+def _find_equilpts(sys, points, params=None):
+    equilpts = []
+    for i, x0 in enumerate(points):
+        # Look for an equilibrium point near this point
+        xeq, ueq = find_eqpt(sys, x0, 0, params=params)
 
+        if xeq is None:
+            continue            # didn't find anything
 
+        # See if we have already found this point
+        seen = False
+        for x in equilpts:
+            if np.allclose(np.array(x), xeq):
+                seen = True
+        if seen:
+            continue
+
+        # Save a new point
+        equilpts += [xeq.tolist()]
+
+    return equilpts
+
+
+def _make_points(pointdata, gridspec, gridtype):
+    # Check to see what type of data we got
+    if isinstance(pointdata, np.ndarray) and gridtype is None:
+        pointdata = np.atleast_2d(pointdata)
+        if pointdata.shape[1] == 2:
+            # Given a list of points => no action required
+            return pointdata, None
+
+    # Utility function to parse (and check) input arguments
+    def _parse_args(defsize):
+        # if not isinstance(pointdata, (list, tuple)) or len(pointdata) != 4:
+        #     raise ValueError("invalid grid data specification")
```

Comment:
```suggestion
```
This can be deleted, or the check in the comment can be expanded to handle case of `isinstance(pointdata, np.ndarray)`.